app/services/VatRegistrationService.scala

```diff
@@ -82,4 +82,7 @@ class VatRegistrationService @Inject()(val s4LService: S4LService,
   def storePartialVatScheme(regId: String, partialVatScheme: JsValue)(implicit hc: HeaderCarrier): Future[JsValue] =
     vatRegConnector.upsertVatScheme(regId, partialVatScheme)
 
+  def isSoleTrader(implicit profile: CurrentProfile, hc: HeaderCarrier): Future[Boolean] = {
```

Comment:
It would be better for this to just return the party type i.e. 

`def partyType(...): Future[PartyType] = ???`

That way the method will be more useful in the coming months as we add more business entities and we'll be able to use partial functions to keep the code compact. e.g.

`vatRegistrationService.partyType map {
  case Individual => doSomething()
  case UkCompany => doSomethingElse()
}`

Reply:
I have refactored code, and now using `def partyType` where ever `isSoleTrader` was being used